contracts/schemes/ContributionRewardExt.sol

```diff
@@ -0,0 +1,452 @@
+        }
+    }
+
+    /**
+    * @dev redeemNativeTokenFromExtContract redeem reward for proposal
+    * @param _proposalId the ID of the voting in the voting machine
+    * @param _beneficiary the beneficiary to mint tokens to.
+    * @param _amount the tokens amount to mint
+    */
+    function redeemNativeTokenFromExtContract(bytes32 _proposalId, address _beneficiary, uint256 _amount)
+    public
+    onlyRewarder
+    {
+        ContributionProposal storage proposal = organizationProposals[_proposalId];
+        require(proposal.executionTime != 0);
+        //this will ensure sum zero of reputation.
+        proposal.nativeTokenRewardLeft = proposal.nativeTokenRewardLeft.sub(_amount);
+
+        if (_amount > 0) {
+            address(avatar.nativeToken()).safeTransfer(_beneficiary, _amount);
+            emit RedeemNativeToken(address(avatar), _proposalId, _beneficiary, _amount);
+        }
+    }
+
+    /**
+    * @dev redeemEtherFromExtContract redeem reward for proposal
+    * @param _proposalId the ID of the voting in the voting machine
+    * @param _beneficiary the beneficiary to send eth to.
+    * @param _amount eth amount to send
+    */
+    function redeemEtherFromExtContract(bytes32 _proposalId, address payable _beneficiary, uint256 _amount)
+    public
+    onlyRewarder
+    {
+        ContributionProposal storage proposal = organizationProposals[_proposalId];
+        require(proposal.executionTime != 0);
+        //this will ensure sum zero of reputation.
+        proposal.ethRewardLeft = proposal.ethRewardLeft.sub(_amount);
+
+        if (_amount > 0) {
+            _beneficiary.transfer(_amount);
+            emit RedeemEther(address(avatar), _proposalId, _beneficiary, _amount);
+        }
+    }
+
+    /**
+    * @dev redeemExternalTokenFromExtContract redeem reward for proposal
+    * @param _proposalId the ID of the voting in the voting machine
+    * @param _beneficiary the beneficiary to send the external token to.
+    * @param _amount the amount of external token to send
+    */
+    function redeemExternalTokenFromExtContract(bytes32 _proposalId, address _beneficiary, uint256 _amount)
+    public
+    onlyRewarder {
+        ContributionProposal storage proposal = organizationProposals[_proposalId];
+        require(proposal.executionTime != 0);
+        //this will ensure sum zero of reputation.
+        proposal.externalTokenRewardLeft = proposal.externalTokenRewardLeft.sub(_amount);
+
+        if (proposal.externalToken != IERC20(0)) {
+            if (_amount > 0) {
+                address(proposal.externalToken).safeTransfer(_beneficiary, _amount);
+                emit RedeemExternalToken(address(avatar), _proposalId, _beneficiary, _amount);
+            }
+        }
+    }
+
+    /**
+    * @dev redeem rewards for proposal
+    * @param _proposalId the ID of the voting in the voting machine
+    * @param _whatToRedeem whatToRedeem array:
```

Comment:
this is an helper function to enable redeem of all in one tx.
the `fine control` is needed to avoid reverts for the case the dao does not hold all the rewards of the time of the call .
and there was also a need to maintain the flexibility to call each one of it separately.